### DISCUSSION TITLE
Roles. actions, and resource relationship mapping

### DISCUSSION THREAD

**Discussion by JustinYeoh93** [argoproj/argo-cd#8449] [Q&A]
Hi all,

I'm reading through the docs and trying to figure out what kind of permissions do I want to give my developers.
Currently, I want them to be able to do sync and rollbacks.

I thought `sync` was sufficient but it would seem that I need to turn on `update` on the application resource as well.
I'm not too comfortable as I'm not sure what other things in the application level does the `update` permission give.

Are there any docs that maps this out?
If not, what other things am I allowing a user with `update` application be able to do?

**Comment by rishabh625** [comment]
sync and rollback can be performed by user  if it has sync permission , update permission is not required , you can validate what all things user can perform 

eg: argocd admin settings rbac can admin sync application 'default/test' --namespace argocd
would return yes if user is able to do so

Follow this guide to configure user roles
https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/rbac.md

example to allow sync 

data:
  policy.csv: |
    p, role:dev, applications, sync, \*/\*, allow
    g, alice, role:dev

Any User(alice - in above example)  with role dev can perform sync/rollback on any project/application (\*/\*)
